src/mjolnir/ferry_connections.cc

```diff
@@ -225,13 +225,27 @@ bool ShortFerry(const uint32_t node_index,
   uint64_t wayid = 0;
   bool short_edge = false;
   for (const auto& edge : bundle.node_edges) {
-    // Check ferry edge. If the end node has a non-ferry edge check
-    // the length of the edge
+    // Check ferry edge.
     if (edge.first.attributes.driveable_ferry) {
       uint32_t endnode =
           (edge.first.sourcenode_ == node_index) ? edge.first.targetnode_ : edge.first.sourcenode_;
       auto end_node_itr = nodes[endnode];
       auto bundle2 = collect_node_edges(end_node_itr, nodes, edges);
+
+      // If we notice that either the end node or source node is a connection to another ferry edge,
+      // be cautious and assume this isn't a short edge.
+      for (const auto& edge2 : bundle.node_edges) {
+        if (edge2.first.llindex_ != edge.first.llindex_ && edge2.first.attributes.driveable_ferry) {
+          return false;
+        }
+      }
+      for (const auto& edge2 : bundle2.node_edges) {
+        if (edge2.first.llindex_ != edge.first.llindex_ && edge2.first.attributes.driveable_ferry) {
+          return false;
+        }
+      }
+
+      // If the end node has a non-ferry edge check the length of the edge
       if (bundle2.node.non_ferry_edge_) {
```

Comment:
The code previously had this if statement here as the "just one edge" check, but it wasn't sufficient. If we consider this case (used in the unit test described below):
```
          A--B--C--F--D--E--G
                   |
                   N
```
where `CF` and `FD` are ferry edges and `FN` is *not* a ferry edge. Suppose that `ShortFerry` is analyzing `CF` and `bundle2` is created at `F`. Then `bundle2.node.non_ferry_edge_` will be `true` since `FN` exists.

Reply:
Haha, guess it was working in most cases because not typical for a 'traditional'  ferry to have a non-ferry edge somewhere in the middle